### PR TITLE
feat(cli): support CloudFormation simplified resource import

### LINKED ISSUE #28060: cli: support CloudFormation simplified resource import
### Describe the feature

On Nov 17 2023, a new feature was announced: AWS CloudFormation simplifies resource import with a new parameter for ChangeSets.

https://aws.amazon.com/about-aws/whats-new/2023/11/aws-cloudformation-import-parameter-changesets/

Summary:

AWS CloudFormation's new ImportExistingResources parameter for CreateChangeSet API automatically imports existing resources in the template during deployments. It identifies resources using custom names, avoiding errors from naming conflicts. This reduces manual effort as a separate resource name document is no longer needed. ImportExistingResources allows importing existing resources and deploying changes in a single ChangeSet.

### Use Case

Automatically import exsting resources with the same physical name, such as S3 bucket, DDB table, etc, during a CFn deployment. 

Because resource import is a vital feature for CDK users e.g. to refactor a construct tree, it would benefit many potential users if cdk natively support it.

### Proposed Solution

This feature introduced a new request parameter [`ImportExstingResources`](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_CreateChangeSet.html#:~:text=ImportExistingResources) to the `CreateChangeSet` API. So I guess we can add a CLI option like `--import-exsting-resources: boolean` to `cdk deploy` command to enable the new feature.

The API is called here:
https://github.com/aws/aws-cdk/blob/c66e197f6f8840da6475383dbf2421c3b06ea417/packages/aws-cdk/lib/api/deploy-stack.ts#L410-L418

### Other Information

_No response_

### Acknowledgements

- [x] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.110.0

### Environment details (OS name and version, etc.)

macOS

### PR DESCRIPTION
**Pull request by tmokmss**
Closes #28060

This PR adds a CLI option `--import-exsting-resources: boolean` to cdk deploy command and pass it to `createChangeSet` API call.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by tmokmss** [comment]
Exemption Request: I added a CLI integ test instead of framework-integ.

**Comment by tmokmss** [comment]
@vinayak-kukreja Hi, by any chance, can you run a CLI integ test pipeline and add an integ exempt label to keep this PR open?

File: packages/aws-cdk/lib/cli.ts [hunk 1]
```diff
@@ -547,16 +548,19 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
             if (args.changeSetName) {
               throw new Error('--change-set-name cannot be used with method=direct');
             }
+            if (args.importExistingResources !== undefined) {
+              throw new Error('--import-existing-resources cannot be used with method=direct');
+            }
```
Comment (msambol) [inline packages/aws-cdk/lib/cli.ts:553]: possible to add a unit for this ?

Comment (tmokmss) [reply packages/aws-cdk/lib/cli.ts:553]: Afaik there's no unit test for cli.ts, so I added a test case in cli-integ.

Comment (tmokmss) [reply packages/aws-cdk/lib/cli.ts:553]: But thanks anyway. I forgot that `args.importExistingResources` is  by default `false` not `undefined`, so there was a bug here...

File: packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts [hunk 2]
```diff
@@ -280,6 +280,65 @@ integTest('deploy without execute a named change set', withDefaultFixture(async
...
+  const stackName = 'iam-test';
+  await expect(fixture.cdkDeploy(stackName, {
+    options: ['--import-existing-resources', '--method=direct'],
+  })).rejects.toThrow('exited with error');
+
+  // Ensure stack was not deployed
+  await expect(fixture.aws.cloudFormation('describeStacks', {
+    StackName: fixture.fullStackName(stackName),
+  })).rejects.toThrow('does not exist');
+}));
```
Comment (msambol) [inline packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts:340]: thanks for adding this 👍🏼

**Comment by msambol** [comment]
LGTM 👍🏼 Thanks for adding this!

**Review by msambol — APPROVED** [review]
LGTM. Thanks 👍🏼

**Pull request closed without merging**